src/test/java/com/zendesk/maxwell/producer/InflightMessageListTest.java

```diff
@@ -54,4 +59,36 @@ public void testOutOfOrderComplete() {
 		ret = list.completeMessage(p1);
 		assertEquals(p3, ret);
 	}
+
+	@Test
+	public void testAddMessageWillWaitWhenCapacityIsFull() throws InterruptedException {
+		AddMessage addMessage = new AddMessage();
+		new Thread(addMessage).start();
+		assertThat("Should never exceed capacity", list.size(), lessThanOrEqualTo(capacity));
+
+		long wait = 500;
+		Thread.sleep(wait);
+		list.completeMessage(p1);
+
+		Thread.sleep(100);
```

Comment:
The first `sleep()` is unavoidable, but you could replace this one with a `join()` on the `AddMessage` thread.

Reply:
👍 nice, good to know